crates/solver/src/solver/http_solver/instance_creation.rs

```diff
@@ -101,33 +111,30 @@ impl InstanceCreator {
         // slow down the solver and the solver can estimate them on its own.
         let price_estimates = external_prices.into_http_solver_prices();
 
-        let fee_connected_tokens = if filter_non_fee_connected_orders {
-            // For the optimization HTTP solver to run correctly we need to be
-            // sure that there are no isolated islands of tokens without
-            // connection between them. Ideally, this filtering **should not be
-            // needed** and done in the optimization solvers themselves, since
-            // it is logic specific to those solvers.
-            compute_fee_connected_tokens(&amms, self.native_token)
-        } else {
-            // For external solvers assume all tokens are connected to the fee
-            // token as they may use additional internal liquidity that we don't
-            // know about.
-            tokens.iter().copied().collect()
-        };
         let gas_model = GasModel {
             native_token: self.native_token,
             gas_price,
         };
 
+        // Some solvers require that there are no isolated islands of tokens unconnected to the
+        // native token.
```

Comment:
This comment currently suggests to me that the final filtered `Instance` should rather have its `tokens` updated than the `orders`. I had to `git blame` Nick's original change to make sure that we only care about filtering orders and not tokens or amms.

Reply:
I think only matters for orders.

Reply:
Changed comment to talk about orders.